libvast/src/system/disk_monitor.cpp

```diff
@@ -34,21 +36,48 @@ std::shared_ptr<caf::detail::scope_guard<Fun>> make_shared_guard(Fun f) {
 
 } // namespace
 
+caf::expected<void> validate(const disk_monitor_config& config) {
```

Comment:
```suggestion
caf:error validate(const disk_monitor_config& config) {
```

This might be more natural. In fact, `expected<void>` is a specialization that just wraps an `error`.